### PR TITLE
feat: add new endpoint to check translation status

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-1826

# Goal

- Provide endpoint to check translation status
- TBD... provide user feedback accordingly

A new endpoint was introduced to get translatable status

```
GET /tao/Translation/status?id=https://backoffice.ngs.test/ontologies/tao.rdf%23i673cb1b150527202411191641378f8c0526

{
    "success": true,
    "data": {
        "uri": "https://backoffice.ngs.test/ontologies/tao.rdf#i673cb1b150527202411191641378f8c0526",
        "type": "http://www.tao.lu/Ontologies/TAOTest.rdf#Test",
        "isReadyForTranslation": true,
        "isEmpty": true
    }
}
```

# Related PRs

- https://github.com/oat-sa/extension-tao-itemqti/pull/2627
- https://github.com/oat-sa/extension-tao-testqti/pull/2530

Screenshot 2024-11-20 at 15 14 18

### CODE REVIEW

**Review by jsconan — APPROVED** [review]
Quick code review. for the frontend part only. Looking good, nice changes!

**Review Checklist**
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code respects code style rules
- [x] New code respects best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Review by shpran — APPROVED** [review]
Great job!
LGTM 👍🏻

File: models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        return $status;
+    }
+
+    private function isReadyForTranslation(core_kernel_classes_Resource $resource): bool
+    {
+        $translationStatusProperty = $this->ontology->getProperty(TaoOntology::PROPERTY_TRANSLATION_STATUS);
+
+        if (!$translationStatusProperty) {
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php:90]: I want to cover the case where some legacy item/test is passed in the API

File: models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        return $translationStatusPropertyValue->getUri() === TaoOntology::PROPERTY_VALUE_TRANSLATION_STATUS_READY;
+    }
+
+    private function getLanguageUri(core_kernel_classes_Resource $resource): ?string
+    {
+        $languageProperty = $this->ontology->getProperty(TaoOntology::PROPERTY_LANGUAGE);
+
+        if (!$languageProperty) {
```
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php:107]: I want to cover the case where some legacy item/test is passed in the API

File: models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (shpran) [reply models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php:90]: But here you are not touching item/test property value - you're just receiving a common property object from DB.
This one exists every time (if it presents in tao.rdf + synched).

File: models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php]: Thanks! Done!

File: models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrielfs7) [reply models/classes/Translation/Service/ResourceTranslatableStatusRetriever.php]: Thanks! Done!

**Pull request merged**